lasagne/layers/shape.py

```diff
@@ -300,3 +302,69 @@ def get_output_for(self, input, **kwargs):
         return padding.pad(input, self.width, self.val, self.batch_ndim)
 
 pad = PadLayer  # shortcut
+
+
+class SliceLayer(Layer):
+    """
+    Slices the input at a specific axis and at specific indices.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+
+    indices : int, list, tuple or slice instance
+        If an ``int``, selects a single element from the given axis, dropping
+        the axis. If a slice, selects all elements in the given range, keeping
+        the axis. If a list or a tuple of ``start, stop, step``, it will be
+        converted to a slice.
+
+    axis : int
+        Specifies the axis from which the indices are selected.
+
+    Examples
+    --------
+    >>> from lasagne.layers import SliceLayer, InputLayer, get_output
+    >>> in_shp = (2, 3, 2)
+    >>> l_inp = InputLayer(in_shp)
+    >>> l_slice_ax0 = SliceLayer(l_inp, axis=0, indices=0)
+    >>> l_slice_ax1 = SliceLayer(l_inp, axis=1, indices=(1, 3, 1))
+    >>> l_slice_ax2 = SliceLayer(l_inp, axis=-1, indices=-1)
+    >>> x = np.arange(np.prod(in_shp)).reshape(in_shp).astype('float32')
+    >>> l_in = InputLayer((2, 3, 4))
+    >>> SliceLayer(l_in, indices=0, axis=1).output_shape
+    ... # equals input[:, 0]
+    (2, 4)
+    >>> SliceLayer(l_in, indices=slice(0, 1), axis=1).output_shape
+    ... # equals input[:, 0:1]
+    (2, 1, 4)
+    >>> SliceLayer(l_in, indices=slice(-2, None), axis=-1).output_shape
+    ... # equals input[..., -2:]
+    (2, 3, 2)
+    """
+    def __init__(self, incoming, indices, axis=-1, **kwargs):
+        super(SliceLayer, self).__init__(incoming, **kwargs)
+        if isinstance(indices, (int, slice)):
+            self.slice = indices
+        elif isinstance(indices, (list, tuple)) and len(indices) == 3:
+            self.slice = slice(*indices)
```

Comment:
Okay, cool! Feel free to go ahead and remove them then :) Don't forget to change the docstring (i.e., delete the last sentence and remove `list` and `tuple`) and the test (i.e., change the tuple to a slice). This will allow us to add lists for fancy indexing later if we wanted to, or for multi-axis slicing.

PS: Another fun way for the slice layer syntax would be:
```python
l_in = InputLayer((2,3,4))
l1 = SliceLayer(l_in)[:,-1]
```
That would probably raise some eyebrows, though.